src/external_copy/serializer_nortti.cc

```diff
@@ -18,7 +18,7 @@ void SerializerDelegate::ThrowDataCloneError(Local<String> message) {
 
 auto SerializerDelegate::GetSharedArrayBufferId(
 		Isolate* /*isolate*/, Local<SharedArrayBuffer> shared_array_buffer) -> Maybe<uint32_t> {
-	auto result = Nothing<uint32_t>();
+	auto result = Just<uint32_t>(0);
```

Comment:
These changes are meaningfully different. In the case the nested run barrier fails this will return a successful value instead of a failure. What compilation warning are you seeing? I didn't see any.

https://github.com/v8/v8/blob/fb9a925cdc18e8277123212b8c7bc270c483b8b9/include/v8-value-serializer.h#L92-L97

> If not, a suitable exception should be thrown and Nothing<bool>() returned.

`Nothing<bool>()` and `Just(false)` are not the same.

Reply:
When compiling with `-fsanitize=address`:
```
In file included from /home/orgads/.cache/node-gyp/20.10.0/include/node/v8-object.h:9,
                 from /home/orgads/.cache/node-gyp/20.10.0/include/node/v8-array-buffer.h:13,
                 from /home/orgads/.cache/node-gyp/20.10.0/include/node/v8.h:24,
                 from ../src/external_copy/external_copy.h:2,
                 from ../src/external_copy/serializer.h:2,
                 from ../src/external_copy/serializer_nortti.cc:1:
In function ‘v8::Maybe<T> v8::Nothing() [with T = bool]’,
    inlined from ‘virtual v8::Maybe<bool> ivm::detail::SerializerDelegate::WriteHostObject(v8::Isolate*, v8::Local<v8::Object>)’ at ../src/external_copy/serializer_nortti.cc:37:29:
/home/orgads/.cache/node-gyp/20.10.0/include/node/v8-maybe.h:112:19: warning: ‘*(unsigned char*)((char*)&<unnamed> + offsetof(v8::Maybe<bool>,v8::Maybe<bool>::value_))’ is used uninitialized [-Wuninitialized]
  112 |   return Maybe<T>();
      |                   ^
/home/orgads/.cache/node-gyp/20.10.0/include/node/v8-maybe.h: In member function ‘virtual v8::Maybe<bool> ivm::detail::SerializerDelegate::WriteHostObject(v8::Isolate*, v8::Local<v8::Object>)’:
/home/orgads/.cache/node-gyp/20.10.0/include/node/v8-maybe.h:112:10: note: ‘<anonymous>’ declared here
  112 |   return Maybe<T>();
      |          ^~~~~~~~~~
In function ‘v8::Maybe<T> v8::Nothing() [with T = unsigned int]’,
    inlined from ‘virtual v8::Maybe<unsigned int> ivm::detail::SerializerDelegate::GetSharedArrayBufferId(v8::Isolate*, v8::Local<v8::SharedArrayBuffer>)’ at ../src/external_copy/serializer_nortti.cc:21:33:
/home/orgads/.cache/node-gyp/20.10.0/include/node/v8-maybe.h:112:19: warning: ‘<anonymous>.v8::Maybe<unsigned int>::value_’ is used uninitialized [-Wuninitialized]
  112 |   return Maybe<T>();
      |                   ^
/home/orgads/.cache/node-gyp/20.10.0/include/node/v8-maybe.h: In member function ‘virtual v8::Maybe<unsigned int> ivm::detail::SerializerDelegate::GetSharedArrayBufferId(v8::Isolate*, v8::Local<v8::SharedArrayBuffer>)’:
/home/orgads/.cache/node-gyp/20.10.0/include/node/v8-maybe.h:112:10: note: ‘<anonymous>’ declared here
  112 |   return Maybe<T>();
      |          ^~~~~~~~~~
```

Reply:
The compiler warnings are overzealous in this case. The memory is not initialized but this is a conscious design decision by the v8 authors. The proposed change to isolated-vm is incorrect and would swallow exceptions silently.

Reply:
Reverted.